core/src/main/scala/io/delta/sql/parser/DeltaSqlParser.scala

```diff
@@ -156,6 +156,29 @@ class DeltaSqlAstBuilder extends DeltaSqlBaseBaseVisitor[AnyRef] {
       ctx.RUN != null)
   }
 
+  /**
+   * Create a [[OptimizeTableCommand]] logical plan. Example SQL:
+   * {{{
+   *    OPTIMIZE '/path/to/delta/table';
+   *    OPTIMIZE delta.db.deltaTable;
+   *    OPTIMIZE delta.db.deltaTable WHERE partCol = 25;
+   * }}}
+   */
+  override def visitOptimizeTable(ctx: OptimizeTableContext): AnyRef = withOrigin(ctx) {
+    if (ctx.path == null && ctx.table == null) {
+      throw new ParseException("OPTIMIZE command requires a file path or table name.", ctx)
+    }
+    OptimizeTableCommand(
+      Option(ctx.path).map(string),
+      Option(ctx.table).map(visitTableIdentifier),
+      Option(ctx.partitionPredicate).map(partitionPredicate => {
+        // Extract the raw expression text which will be parsed later
+        partitionPredicate.getStart.getInputStream.getText(new Interval(
+          partitionPredicate.getStart.getStartIndex,
+          partitionPredicate.getStop.getStopIndex))
```

Comment:
nit: we can create a method such as 

```
protected def extractRawText(context: org.antlr.v4.runtime.ParserRuleContext): String = {
  ...
}
```
Then we can use this to fix the check constraint later.